custom/staking/src/lib.rs

```diff
@@ -1314,6 +1346,62 @@ decl_module! {
             Self::deposit_event(RawEvent::InvalidatedNominators(caller, expired_nominators));
         }
 
+        /// Enables individual commisions. This can be set only once. Once individual commission
+        /// rates are enabled, there's no going back.  Only Governance committee is allowed to
+        /// change this value.
+        #[weight = SimpleDispatchInfo::FixedOperational(100_000)]
+        fn enable_individual_commissions(origin) {
+            T::RequiredCommissionOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| Error::<T>::NotAuthorised)?;
+
+            // Ensure individual commissions are not already enabled
+            if let Commission::Global(_) = <ValidatorCommission>::get() {
+                <ValidatorCommission>::put(Commission::Individual);
+                Self::deposit_event(RawEvent::IndividualCommissionInEffect);
+            } else {
+                Err(Error::<T>::IndividualCommissionsEnabled)?
+            }
+        }
+
+        /// Changes commission rate which applies to all validators. Only Governance
+        /// committee is allowed to change this value.
+        ///
+        /// # Arguments
+        /// * `new_value` the new commission to be used for reward calculations
+        #[weight = SimpleDispatchInfo::FixedOperational(100_000)]
+        fn set_global_comission(origin, new_value: Perbill) {
+            T::RequiredCommissionOrigin::try_origin(origin)
+                .map(|_| ())
+                .or_else(ensure_root)
+                .map_err(|_| Error::<T>::NotAuthorised)?;
+
+            // Ensure individual commissions are not already enabled
+            if let Commission::Global(old_value) = <ValidatorCommission>::get() {
+                <ValidatorCommission>::put(Commission::Global(new_value));
```

Comment:
It isn't better to check `old_value != new_value` ?

Reply:
Not sure about this. I dont think we reject anywhere if a function is called with the same value again.

Reply:
IMO it doesn't hurt anyone but It is a waste of fee to call the function. Just for a sanity check

Reply:
Added ensure to reject if there's no change.